tests/pos/i14340.scala

```diff
@@ -0,0 +1,10 @@
+object i14340 {
```

Comment:
As this used to fail at runtime, we need to have a test checking if this works at runtime. Tests in `tests/pos` are only checked for successful compilation. You need to move this test to `tests/run/i14340.scala` and make it a runnable program (e.g. by creating a `@main` method).  Also for extra certainty that this actually works as expect you could add a test check file `tests/run/i14340.check` containing the expected console output of the program. In this case it would be simply
```
1
```

Reply:
Thanks! I did that and solution is not working. Will try something else

Reply:
I've found solution, but again I'm not sure about it. I think we nned to wrap called members of Selectable. But it looks not perfect for me

Reply:
I have some doubts if `Dynamic.scala` is really the proper place for the fix. I tried to play with the issue a bit and it seems specific to `reflect.Selectable` rather than to `Selectable` in general. For sure we need more tests here to make sure we won't break something else. Consider cases like this:
* What if we need to deal with nested value classes?
* What if what we call from a `Selectable` should desugar to `applyDynamic` rather than `selectDynamic`?
* What if we provide a custom implementation of `selectDynamic` which doesn't use reflection?

An extended test case could look like this
```scala
class Foo(val value: Int) extends AnyVal
class Bar[A](val value: A) extends AnyVal

class Container1 extends reflect.Selectable

class Container2 extends Selectable:
  def selectDynamic(name: String) = Bar(name)

val cont1 = new Container1:
  val foo = new Foo(1)
  val bar = new Bar(Foo(2))
  def fooFromInt(i: Int) = new Foo(i)

val cont2 = (new Container2).asInstanceOf[Container2 { def qux: Bar[String] }]

@main def Test: Unit =
  println(cont1.foo.value)
  println(cont1.bar.value)
  println(cont1.fooFromInt(3).value)
  println(cont2.qux.value)
```
with expected output
```
1
2
3
qux
```

Reply:
Thanks a lot. I will think about another approach